distributed/deploy/adaptive.py

```diff
@@ -188,6 +190,17 @@ def should_scale_up(self):
             if self.scheduler.unrunnable and not self.scheduler.workers:
                 return True
 
+            if not all(ws.processing for ws in self.scheduler.workers.values()):
+                return False
+
+            tasks_processing = sum(
+                (len(w.processing) for w in self.scheduler.workers.values()))
+
+            num_workers = len(self.scheduler.workers)
+
+            if tasks_processing <= num_workers:
+                return False
```

Comment:
I think I can convince myself that the `tasks_processing < num_workers` check is redundant given the `all(ws.processing ...)` check.

Reply:
it's definitely sufficient to prompt the correct behavior when the cluster has idle workers, but once those have been spun down, the current logic doesn't distinguish between a resource-constrained cluster with waiting tasks and one with ntasks == nworkers. This was confirmed by my experiments, where the cluster successfully scaled down to (ntasks) and then shot back up to `adaptive.maximum` a few heartbeats later, and continued oscilating between these two until all tasks completed.

I don't see a way to do this with `all(ws.processing ...)` but if there is then the performance benefit would be significant.